server.go

```diff
@@ -1944,6 +1953,33 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 	return nil
 }
 
+// SetSendCompressor sets the compressor that will be used when sending
+// RPC payload back to the client. It may be called at most once, and must not
+// be called after any event that causes headers to be sent (see SetHeader for
+// a complete list). Provided compressor is used when below conditions are met:
+//
+//   - compressor is registered via encoding.RegisterCompressor
+//   - compressor name exists in the client advertised compressor names sent in
+//     grpc-accept-encoding metadata.
+//
+// The context provided must be the context passed to the server's handler.
+//
+// The error returned is compatible with the status package.  However, the
+// status code will often not match the RPC status as seen by the client
+// application, and therefore, should not be relied upon for this purpose.
+func SetSendCompressor(ctx context.Context, name string) error {
+	stream, ok := ServerTransportStreamFromContext(ctx).(*transport.Stream)
```

Comment:
I didn't add it to the interface to avoid breaking existing implementations, though I just noticed it's experimental which should be okay I guess. We need to add two methods `ClientAdvertisedCompressors` and `SetSendCompress` in the interface to avoid type assertion